Group-5-Project-3/Backend
# Feature/trail image service test

### Summary:
This PR adds unit tests for the `TrailImageService` class to enhance test coverage and validate core functionality.

### Changes Made:
1. Added unit tests for:
   - `getImagesByTrailId()`
   - `getImagesByUserId()`
   - `saveTrailImage()`
   - `getTrailImageById()`
   - `deleteTrailImage()`
2. Mocked dependencies using Mockito.
3. Verified core logic and edge cases.

### How to Test:
- Run `mvn test` to ensure all tests pass.
- Review the updated file at `src/test/java/com/project3/project3/service/TrailImageServiceTest.java`.
